ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,178 @@
+    )
+
+    date = forms.DateField(
+        label='Date of reservation (mm/dd/yy)',
+        widget=forms.DateInput(attrs={'placeholder': '02/28/18'}),
+    )
+
+    starttime = forms.TimeField(
+        label='Starting time of reservation (xx:xx)',
+    )
+
+    endtime = forms.TimeField(
+        label='Ending time of reservation (xx:xx)',
+    )
+
+    disclaimer_agreement = forms.BooleanField(
+        label='I agree with the above statement.',
+        error_messages={
+            'required': 'You must agree to our policies.',
+        },
+    )
+
+    def clean_verify_contact_email(self):
+        email = self.cleaned_data.get('contact_email')
+        verify_contact_email = self.cleaned_data.get('verify_contact_email')
+
+        if email and verify_contact_email:
+            if email != verify_contact_email:
+                raise forms.ValidationError("Your emails don't match.")
+        return verify_contact_email
+
+
+@calnet_required
+def request_reservation(request):
+    calnet_uid = request.session['calnet_uid']
+    status = 'new_request'
+
+    # ensure we can even find them in university LDAP
+    # (alumni etc. might not be readable in LDAP but can still auth via CalNet)
+    if not user_attrs_ucb(calnet_uid):
+        return render(
+            request,
+            'lab_reservations/lab_reservations/cant-find-in-ldap.html',
+            {
+                'calnet_uid': calnet_uid,
+                'title': 'Unable to read account information',
+            },
+        )
+
+    if request.method == 'POST':
+        form = RequestForm(request.POST)
+        if form.is_valid():
+            req = NewReservationRequest(
+                real_name=form.cleaned_data['real_name'],
+                contact_email=form.cleaned_data['contact_email'],
+                student_group=form.cleaned_data['student_group'],
+                reason=form.cleaned_data['reason'],
+                date=form.cleaned_data['date'],
+                starttime=form.cleaned_data['starttime'],
+                endtime=form.cleaned_data['endtime'],
+            )
+
+            send_request_to_officers(req)
+            send_request_confirmation(req)
+
+            return render(
+                request,
+                'lab_reservations/pending.html',
+            )
+
+    else:
+        form = RequestForm()
+
+    return render(
+        request,
+        'lab_reservations/index.html',
+        {
+            'form': form,
+            'status': status,
+            'title': 'Request the OCF Lab Space',
+        },
+    )
+
+
+def reservation_requested(request):
+    return render(request, 'lab_reservations/pending.html', {'title': 'Reservation request successful'})
+
+
+def send_request_to_officers(request):
+    body = JINJA_MAIL_ENV.get_template(
+        'lab_reservations/mail_templates/officer_notification.jinja',
+    ).render(request=request)
+    send_mail('officers@berkeley.edu', 'New Lab Reservation Request', body)
```

Comment:
I still think it's a better idea to just have this get sent to bod@ immediately with the requestor set appropriately, without going through officers@ first, and to integrate that "feature" in right now as opposed to waiting for someone to do it in the future.

Reply:
totally agreed -- although keep in mind i think bod@ has a special rule where requestors aren't emailed about replies (not sure if it's still there or not, but we tried implementing it after some embarrassing accidental reply to a bod ticket). might still be okay for this use-case (e.g. it can be moved to `help` after approval or whatever), just wanted to mention.

Reply:
If that's the case, it may be a better long-term option to repurpose the lending queue as the requests queue. Aside from borrowing stuff and asking to reserve the lab, I imagine in the future we'll also want to have an HPC request form, and having a single requests queue would be a good way to manage all of that together.

Reply:
My most recent commit tackles this issue. Instead of an email to officers, instead a ticket is created in the bod queue.